jaxrs/src/main/java/com/facebook/airlift/jaxrs/JerseyAuthorizationFilter.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2010 Proofpoint, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.airlift.jaxrs;
+
+import com.facebook.airlift.http.server.AuthorizationResult;
+import com.facebook.airlift.http.server.Authorizer;
+import com.facebook.airlift.http.server.HttpServerConfig;
+import com.google.common.collect.ImmutableSet;
+import com.google.inject.Inject;
+
+import javax.annotation.security.RolesAllowed;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.container.ResourceInfo;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.Provider;
+
+import java.security.Principal;
+import java.util.Optional;
+import java.util.Set;
+
+import static java.lang.String.format;
+
+@Provider
+public class JerseyAuthorizationFilter
```

Comment:
just a thought. do we need to add `Jersey` in the class name. ?

Reply:
This AuthorizationFilter is used for Jersey resources and is not able to handle async HTTP endpoints (_i.e._ Jetty servlets like [AsyncPageTransportServlet](https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/server/AsyncPageTransportServlet.java)). We'll need a `JettyAuthorizationFilter` to do that.

Therefore, we'll want to differentiate the two types of AuthorizationFilter from their names. If you have better idea for naming please let me know :)

Reply:
Given that the package name already has `jaxrs` we should be fine with just saying `AuthorizationFilter`